### ISSUE TITLE
Have stack just copy unchanged binaries when updating to new LTS

### ISSUE THREAD

**Issue by wolftune** [commercialhaskell/stack#771]
Apologies if somehow this is already happening and I missed it. It would be smoothest and fastest to update to newer LTS of Stackage if stack knew to check the version differences and just copy compiled binaries from previous LTS for unchanged items and only compile updated ones. (Maybe to save space could even just link, although that would make each LTS directory less independent, would cause breakage if older LTS directories were purged, although there could be a function that would copy over linked items prior to purging old LTS collections…)

I understand if most things are changing too often for this to work smoothly enough, like dependencies of dependencies etc. — I suppose it'd have to check and recompile all items that had any of their dependencies change.

If this isn't practical, I understand. But maybe it's feasible…

**Comment by snoyberg** [comment]
We've had discussions about improving this situation, but it won't be a simple copy. In theory reregistering packages from one database to the other would work, but there's the problem that when a user garbage collects an old snapshot the new snapshot would be broken. We can also consider having one massive package database that everything gets installed to and then take views of that, but (1) that requires GHC 7.10 and newer, and (2) Cabal doesn't have the necessary support yet.

tl;dr: There are lots of avenues we can try, but we haven't attacked any of them yet.

**Comment by 3noch** [comment]
This is starting down the `nix` road (it's a nice road, after all). @snoyberg Is there an issue discussing the kinds of things you guys want to try? E.g. views of one mega-db, downloading bindists instead of sdists, etc.

**Comment by snoyberg** [comment]
No, I've been waiting to see the results of the cabal gsoc to add this
support top the library.

**Comment by snoyberg** [comment]
I've opened #878 about this topic. I'm going to close this issue to move discussion over there. If there's something covered here that's not covered there, please say so.
